extensions/ql-vscode/src/databaseFetcher.ts

```diff
@@ -61,6 +64,76 @@ export async function promptImportInternetDatabase(
 
 }
 
+/**
+ * Prompts a user to fetch a database from GitHub.
+ * User enters a GitHub repository and then the user is asked which language
+ * to download (if there is more than one)
+ *
+ * @param databaseManager the DatabaseManager
+ * @param storagePath where to store the unzipped database.
+ */
+export async function promptImportGithubDatabase(
+  databaseManager: DatabaseManager,
+  storagePath: string,
+  credentials: Credentials,
+  progress: ProgressCallback,
+  token: CancellationToken,
+  cli?: CodeQLCliServer
+): Promise<DatabaseItem | undefined> {
+  progress({
+    message: 'Choose repository',
+    step: 1,
+    maxStep: 2
+  });
+  const githubRepo = await window.showInputBox({
+    title: 'Enter a GitHub repository in the format <owner>/<repo> (e.g. github/codeql)',
+    placeHolder: '<owner>/<repo>',
+    ignoreFocusOut: true,
+  });
+  if (!githubRepo) {
+    return;
+  }
+
+  if (!REPO_REGEX.test(githubRepo)) {
+    throw new Error(`Invalid GitHub repository: ${githubRepo}`);
+  }
+
+  const databaseUrl = await convertGithubNwoToDatabaseUrl(githubRepo, credentials, progress);
+  if (databaseUrl) {
+    const octokit = await credentials.getOctokit();
+    /** 
+     * A token object returned by `octokit.auth()`. It's undocumented, but looks something like this:
+     * {
+     *   token: 'xxxx',
+     *   tokenType: 'oauth',
+     *   type: 'token',
+     * }
+     * We only need the actual token string.
+     */
+    const octokitToken = await octokit.auth() as { token: string };
+    if (!octokitToken.token) {
+      // Just print a generic error message for now. Ideally we could show more debugging info, like the
+      // octokit object, but that would expose a user token.
+      throw new Error('Unable to get GitHub token.');
+    }
+    const item = await databaseArchiveFetcher(
+      databaseUrl,
+      { 'Accept': 'application/zip', 'Authorization': `Bearer ${octokitToken.token}` },
+      databaseManager,
+      storagePath,
+      progress,
+      token,
+      cli
+    );
+    if (item) {
+      await commands.executeCommand('codeQLDatabases.focus');
+      void showAndLogInformationMessage('Database downloaded and imported successfully.');
+    }
+    return item;
```

Comment:
Minor: Move this up into the `if(item)` case, so it's more obvious what's going on. The `return` at the end of the function will handle all the fall-through cases where we didn't return early.